<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Peat by jhhung</title>
    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/pygment_trac.css">
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js"></script>
    <script src="javascripts/main.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">

  </head>
  <body>

      <header>
        <h1>Peat</h1>
        <p>An ultra fast and accurate paired-end adapter trimmer that needs no a priori adapter sequences.</p>
      </header>

      <div id="banner">
        <span id="logo"></span>

        <a href="https://github.com/jhhung/PEAT" class="button fork"><strong>View On GitHub</strong></a>
        <div class="downloads">
          <span>Downloads:</span>
          <ul>
            <li><a href="https://github.com/jhhung/PEAT/zipball/master" class="button">ZIP</a></li>
            <li><a href="https://github.com/jhhung/PEAT/tarball/master" class="button">TAR</a></li>
          </ul>
        </div>
      </div><!-- end banner -->

    <div class="wrapper">
      <nav>
        <ul></ul>
      </nav>
      <section>
        <h1>
<a name="peat" class="anchor" href="#peat"><span class="octicon octicon-link"></span></a>PEAT</h1>

<p>PEAT, a fast and highly accurate adapter-trimming algorithm, designed specifically for paired-end sequencing. PEAT requires no a priori adapter sequence and significantly outperforms other widely used adapter trimmers in both simulated and real life paired-end sequencing libraries. Moreover, in order to process single-end sequencing data, we also provide single-end trimmer in PEAT.</p>

<p>PEAT is released under GPLv2 with additional restriction so that is only applicable to individuals and non-profits and that any for-profit company must purchase a different license.</p>

<h2>
<a name="install" class="anchor" href="#install"><span class="octicon octicon-link"></span></a>INSTALL</h2>

<p>*Only 64 bits systems are able to compile and run PEAT. </p>

<h4>
<a name="run-the-binary-directly-without-installation" class="anchor" href="#run-the-binary-directly-without-installation"><span class="octicon octicon-link"></span></a>Run the binary directly without installation</h4>

<p>Try the precompiled binaries first, most of the linux systems should be able to run PEAT without any troubles.</p>

<p>Please find the binary that suits your platform:</p>

<pre><code>bin/PEAT_linux  // for centos/redhat/ubuntu/fedara/...
bin/PEAT_Mac     // for OSX
</code></pre>

<p>Or you can find them in the release tab in this page or at this link:
<a href="https://github.com/jhhung/PEAT/releases">https://github.com/jhhung/PEAT/releases</a></p>

<p>You can rename it to "PEAT" for your convinience.
If unfortunately, none of them works, please see below to build a binary for your box.</p>

<h4>
<a name="install-the-dependencies" class="anchor" href="#install-the-dependencies"><span class="octicon octicon-link"></span></a>Install the dependencies</h4>

<ul>
<li>1.1 Relative recent C++ compiler that support most features of C++11. We recommend <a href="http://gcc.gnu.org/">GCC</a>.</li>
<li>1.2 <a href="http://www.boost.org/users/download/">Boost</a>
</li>
<li>1.3 <a href="http://www.cmake.org/">CMake</a>
</li>
</ul><h4>
<a name="get-the-latest-version-of-the-software" class="anchor" href="#get-the-latest-version-of-the-software"><span class="octicon octicon-link"></span></a>Get the latest version of the software</h4>

<pre><code>git clone git@github.com:jhhung/PEAT.git
</code></pre>

<h4>
<a name="enter-the-folder-peat-and" class="anchor" href="#enter-the-folder-peat-and"><span class="octicon octicon-link"></span></a>Enter the folder PEAT and:</h4>

<ul>
<li>Set enviromental variable "BOOST_ROOT" to the directory of boost if CMake cannot find boost automatically;</li>
<li>Set enviromental variable "CC" and "CXX" to the gcc/g++ compiler you want to use. </li>
</ul><pre><code>cmake .
</code></pre>

<h4>
<a name="compile-the-software-by-typing" class="anchor" href="#compile-the-software-by-typing"><span class="octicon octicon-link"></span></a>Compile the software by typing:</h4>

<pre><code>make
</code></pre>

<h2>
<a name="usage" class="anchor" href="#usage"><span class="octicon octicon-link"></span></a>USAGE</h2>

<h4>
<a name="paired-end-adapter-trimming" class="anchor" href="#paired-end-adapter-trimming"><span class="octicon octicon-link"></span></a>Paired-end adapter trimming</h4>

<pre><code>bin/PEAT paired -1 &lt;file&gt; -2 &lt;file&gt; -o &lt;output&gt; -n &lt;num&gt; -l &lt;num&gt; -r &lt;num&gt; -g &lt;num&gt; -a &lt;num&gt;
</code></pre>

<ul>
<li>-1 : The paired_1 input FastQ file.</li>
<li>-2 : The paired_2 input FastQ file.</li>
<li>-o : Prefix for Output file name, stdout by default</li>
<li>-n : Number of thread to use; if the number is larger than the core available, it will be adjusted automatically.</li>
<li>-l : Minimum gene fragment length, i.e. the fragment length for reverse complement check, 30 bp by default</li>
<li>-r : Mismatch rate applied in first stage reverse complement scan, 0.3 by default</li>
<li>-g : Mismatch rate applied in second stage gene portion check, 0.6 by default</li>
<li>-a : Mismatch rate applied in second stage adapter portion check, 0.4 by default</li>
</ul><h4>
<a name="single-end-adapter-trimming-this-funciton-is-adapted-from-httpsgithubcomvsbuffaloscythe" class="anchor" href="#single-end-adapter-trimming-this-funciton-is-adapted-from-httpsgithubcomvsbuffaloscythe"><span class="octicon octicon-link"></span></a>Single-end adapter trimming [this funciton is adapted from: <a href="https://github.com/vsbuffalo/scythe">https://github.com/vsbuffalo/scythe</a>]</h4>

<pre><code>bin/PEAT single -i &lt;file&gt; -a &lt;string&gt; -q &lt;string&gt; -o &lt;output&gt; -n &lt;num&gt;
</code></pre>

<ul>
<li>-i : The input FastQ file.</li>
<li>-a : The adapter sequence, with minimum length of six characters.</li>
<li>-q : The quality type of input FastQ file. (Acceptable quality type : 1.PHRED 2.SANGER 3.SOLEXA 4.ILLUMINA)</li>
<li>-o : Output fastq file, stdout by default</li>
<li>-n : Number of thread to use; if the number is larger than the core available, it will be adjusted automatically.</li>
</ul><h2>
<a name="citing-peat" class="anchor" href="#citing-peat"><span class="octicon octicon-link"></span></a>Citing PEAT</h2>

<ul>
<li>not yet</li>
</ul><h2>
<a name="contact" class="anchor" href="#contact"><span class="octicon octicon-link"></span></a>Contact</h2>

<pre><code>Jui-Hung Hung &lt;juihunghung@gmail.com&gt;
Yun-Lung Li &lt;obigbando@gmail.com&gt;
Jui-Cheng Weng &lt;johnny231949@gmail.com&gt;
</code></pre>
      </section>
      <footer>
        <p>Project maintained by <a href="https://github.com/jhhung">jhhung</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://twitter.com/michigangraham">mattgraham</a></small></p>
      </footer>
    </div>
    <!--[if !IE]><script>fixScale(document);</script><![endif]-->
    
  </body>
</html>